src/Console/SchemaFactoryMakeCommand.php

```diff
@@ -48,19 +48,26 @@ protected function buildModel($output, $model)
 
         $columns = SchemaFacade::connection($model->getConnectionName())->getColumnListing(config('database.connections.'.config('database.default').'.prefix', '').$model->getTable());
         $connection = $model->getConnection();
-
+        
+        $tableName = config('database.connections.'.config('database.default').'.prefix', '').$model->getTable();
+        
+        
         $definition = 'return Schema::object(\''.class_basename($model).'\')'.PHP_EOL;
         $definition .= '            ->properties('.PHP_EOL;
 
         $properties = collect($columns)
-            ->map(static function ($column) use ($model, $connection) {
-                /** @var Column $column */
-                $column = $connection->getDoctrineColumn(config('database.connections.'.config('database.default').'.prefix', '').$model->getTable(), $column);
-                $name = $column->getName();
-                $default = $column->getDefault();
-                $notNull = $column->getNotnull();
-
-                switch (get_class($column->getType())) {
+            ->map(static function ($columnName) use ($model, $connection, $tableName) {
+//                /** @var Column $column */
+//                $column = $connection->getDoctrineColumn(config('database.connections.'.config('database.default').'.prefix', '').$model->getTable(), $column);
+                $columnType = $connection->getSchemaBuilder()->getColumnType(
+                    $tableName,
+                    $columnName
+                );
+                $name = $columnName;
+                $default = $connection->select("SELECT COLUMN_DEFAULT FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_NAME = '$tableName' AND COLUMN_NAME = '$columnName'")[0]->COLUMN_DEFAULT;
```

Comment:
These two lines retrieve the default values and nullability constraints of columns directly from INFORMATION_SCHEMA.
While this approach works with certain databases like MySQL and MariaDB, it is not universally compatible with all database providers supported by Laravel, such as PostgreSQL and SQL Server.
To improve compatibility with various database management systems, it would be better to use the features of Doctrine DBAL, which provides an abstraction for accessing column metadata in a more generic and provider-independent manner.

Reply:
> These two lines retrieve the default values and nullability constraints of columns directly from `INFORMATION_SCHEMA`. While this approach works with certain databases like MySQL and MariaDB, it is not universally compatible with all database providers supported by Laravel, such as PostgreSQL and SQL Server. To improve compatibility with various database management systems, it would be better to use the features of Doctrine DBAL, which provides an abstraction for accessing column metadata in a more generic and provider-independent manner.

@TartanLeGrand Good point, i have changed the code to use the more standard Doctrine function to handle columns:

```php
$tmpColumns = $connection->getSchemaBuilder()->getColumns($tableName);
```

Thanks!